src/views/catalog/wizard/tabs/scripts/components/CloudInitNetworkForm.tsx

```diff
@@ -0,0 +1,221 @@
+          vmDraft.spec.template.spec.volumes = [...otherVolumes, updatedCloudinitVolume];
+        }),
+      ),
+    [cloudInitVolume, isBase64, updateVM, networkData],
+  );
+
+  React.useEffect(() => {
+    if (networkData) {
+      const ipAddreses = networkData.network.config[0].subnets[0].address.join(', ');
+      const hasChanged =
+        ipAddreses !== '' ||
+        networkData.network.config[0].subnets[0].gateway !== '' ||
+        networkData.network.config[0].name !== '';
+      const isValid = validateIpAddresses(ipAddreses);
+
+      const timerId = setTimeout(() => isValid && hasChanged && onUpdateVM(), 100);
+
+      return () => clearTimeout(timerId);
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [networkData, showForm]);
+
+  // delete networkData if user unchecks the checkbox
+  React.useEffect(() => {
+    if (!showForm && templateHasNetworkData) {
+      updateVM((draft) => {
+        const cloudVolume = getVolumes(draft).find((vol) => vol.cloudInitNoCloud);
+        if (cloudVolume?.cloudInitNoCloud?.networkData) {
+          delete cloudVolume.cloudInitNoCloud.networkData;
+          const otherVolumes = getVolumes(draft).filter((vol) => !vol.cloudInitNoCloud);
+          draft.spec.template.spec.volumes = [...otherVolumes, cloudVolume];
+        }
+      });
+    }
+  }, [setNetworkData, showForm, templateHasNetworkData, updateVM]);
+
+  return (
+    <>
+      <FormGroup fieldId="divider">
+        <Divider />
+      </FormGroup>
+
+      <FormGroup fieldId="network-label" label={t('Network data')}></FormGroup>
+      <FormGroup fieldId="custom-network-checkbox">
+        <Checkbox
+          id="custom-network-checkbox"
+          label={t('Add network data')}
+          description={t('check this option to add network data section to the cloud-init script.')}
+          isChecked={showForm}
+          onChange={setShowForm}
+        />
+      </FormGroup>
+      {showForm && (
+        <>
+          <FormGroup
+            label={t('Ethernet name')}
+            fieldId={'ethernet-name'}
+            className="kv-cloudint-advanced-tab--validation-text"
+          >
+            <TextInput
+              value={networkData?.network?.config[0]?.name || ''}
+              type="text"
+              id={'ethernet-name'}
+              onChange={(v) => onFieldChange(CloudInitNetworkFormKeys.NAME, v)}
+            />
+          </FormGroup>
+          <FormGroup
+            label={t('IP Addresses')}
+            fieldId={'address'}
+            className="kv-cloudint-advanced-tab--validation-text"
+            helperText={t('Use commas to separate between IP addresses')}
+            helperTextInvalid={t('Please provide valid IP addresses, separated by comma.')}
+            validated={
+              validateIpAddresses(
+                networkData?.network?.config[0]?.subnets?.[0]?.address?.join(', '),
+              )
+                ? 'default'
+                : 'error'
```

Comment:
you should not stop user from entering arbitrary string if they want to, if you return "error" here you might prevent user from enteting some values they wish to enter